In [3]:
import pandas as pd

df = pd.read_csv("kc_house_data.csv")

# Drop ignored columns
df = df.drop(columns=["id", "date", "zipcode"])

price = df["price"]
features = df.drop(columns=["price"])

stats = pd.DataFrame({
    "Average": features.mean(),
    "Min": features.min(),
    "Max": features.max(),
    "Variance": features.var()
})

stats

,Average,Min,Max,Variance
bedrooms,3.370842,0.0000,3.300000e+01,8.650150e-01
bathrooms,2.114757,0.0000,8.000000e+00,5.931513e-01
sqft_living,2079.899736,290.0000,1.354000e+04,8.435337e+05
sqft_lot,15106.967566,520.0000,1.651359e+06,1.715659e+09
floors,1.494309,1.0000,3.500000e+00,2.915880e-01
waterfront,0.007542,0.0000,1.000000e+00,7.485226e-03
view,0.234303,0.0000,4.000000e+00,5.872426e-01
condition,3.409430,1.0000,5.000000e+00,4.234665e-01
grade,7.656873,1.0000,1.300000e+01,1.381703e+00
sqft_above,1788.390691,290.0000,9.410000e+03,6.857347e+05


The feature with the lowest average is waterfront, with an average value of approximately 0.0075, meaning that very few houses are waterfront properties.

The feature with the highest average is sqft_living, with an average value of approximately 2079.9 square feet.

In [4]:
correlations = features.corrwith(price).sort_values(ascending=False)
correlations


sqft_living      0.702035
grade            0.667434
sqft_above       0.605567
sqft_living15    0.585379
bathrooms        0.525138
view             0.397293
sqft_basement    0.323816
bedrooms         0.308350
lat              0.307003
waterfront       0.266369
floors           0.256794
yr_renovated     0.126434
sqft_lot         0.089661
sqft_lot15       0.082447
yr_built         0.054012
condition        0.036362
long             0.021626
dtype: float64

All features have a positive correlation coefficient with the response variable (price), meaning that increases in these features are generally associated with higher house prices.

The feature with the highest positive correlation with price is sqft_living, with a correlation coefficient of ~0.70, indicating a strong positive relationship between living area and house price.

There were no features with a negative correlation coefficient with the response variable (price). All features showed positive correlation coefficients, indicating that increases in these features are generally associated with higher house prices.